tests/platform_tests/api/test_sfp.py

```diff
@@ -689,8 +689,15 @@ def test_lpmode(self, duthosts, enum_rand_one_per_hwsku_hostname, localhost, pla
                 logger.warning("test_lpmode: Skipping transceiver {} (not applicable for this transceiver type)".format(i))
                 continue
 
+            lpmode_state_pretest = sfp.get_lpmode(platform_api_conn, i)
+            if lpmode_state_pretest is None:
+                logger.warning("test_lpmode: Skipping transceiver {} (not supported on this platform)".format(i))
+                break
+            # This order makes sure lpmode will get restored to pretest value after test
+            lpmode_states_to_be_tested = [not lpmode_state_pretest, lpmode_state_pretest]
+
             # Enable and disable low-power mode on each transceiver
-            for state in [True, False]:
+            for state in lpmode_states_to_be_tested:
```

Comment:
@longhuan-cisco where are we restoring the pretest value `lpmode_state_pretest`

Reply:
as explained in detail in the previous comment, changing the order of lpmode_states_to_be_tested sequence according to value of lpmode_pretest will make sure lpmode state gets restored to the pre-test state at the end of test.

Reply:
ok